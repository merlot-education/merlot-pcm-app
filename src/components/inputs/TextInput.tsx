import React, { useState } from 'react'
import {
  View,
  Text,
  StyleSheet,
  TextInput as RNTextInput,
  TextInputProps,
} from 'react-native'

import { ColorPallet, TextTheme, borderRadius } from '../../theme/theme'

interface Props extends TextInputProps {
  label: string
}

const styles = StyleSheet.create({
  container: {
    marginVertical: 10,
  },
  label: {
    ...TextTheme.label,
    marginBottom: 3,
  },
  textInput: {
    padding: 10,
    borderRadius,
    fontSize: 16,
    backgroundColor: ColorPallet.grayscale.white,
    color: ColorPallet.brand.primary,
    borderWidth: 2,
    borderColor: ColorPallet.baseColors.green,
  },
})

const TextInput: React.FC<Props> = ({ label, ...textInputProps }) => {
  const [focused, setFocused] = useState(false)

  return (
    <View style={styles.container}>
      <Text style={styles.label}>{label}</Text>
      <RNTextInput
        style={[
          styles.textInput,
          focused && { borderColor: ColorPallet.brand.primary },
        ]}
        selectionColor={ColorPallet.brand.primary}
        onFocus={() => setFocused(true)}
        onBlur={() => setFocused(false)}
        {...textInputProps}
      />
    </View>
  )
}

export default TextInput
