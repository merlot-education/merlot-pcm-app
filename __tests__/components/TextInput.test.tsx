import React from 'react'
import { create } from 'react-test-renderer'

import TextInput from '../../src/components/inputs/TextInput'

describe('TextInput Component', () => {
  it('TextInput renders correctly', () => {
    const tree = create(<TextInput label="TextInput Component" />).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
