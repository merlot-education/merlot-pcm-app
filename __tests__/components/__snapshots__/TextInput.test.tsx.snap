// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput Component TextInput renders correctly 1`] = `
<View
  style={
    Object {
      "marginVertical": 10,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#03028A",
        "fontSize": 14,
        "fontWeight": "bold",
        "marginBottom": 3,
      }
    }
  >
    TextInput Component
  </Text>
  <TextInput
    onBlur={[Function]}
    onFocus={[Function]}
    selectionColor="#03028A"
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": undefined,
          "borderRadius": 4,
          "borderWidth": 2,
          "color": "#03028A",
          "fontSize": 16,
          "padding": 10,
        },
        false,
      ]
    }
  />
</View>
`;
